src/overlays/actors/ovl_Obj_Warp2block/z_obj_warp2block.c

```diff
@@ -43,36 +42,305 @@ const ActorInit Obj_Warp2block_InitVars = {
     (ActorFunc)ObjWarp2block_Update,
     (ActorFunc)ObjWarp2block_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA1DF0.s")
+typedef struct {
+  f32 scale;
+  f32 focus;
+  s16 params;
+} Warp2BlockSpawnData;
+
+ Warp2BlockSpawnData sSpawnData[2] = {
+	{ 1.0f, 60.0f, 0x0018 },
+	{ 0.6f, 40.0f, 0x0019 },
+};
+
+f32 sDistances[] = {60.0f, 100.0f, 140.0f, 180.0f, 220.0f, 260.0f, 300.0f, 300.0f};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 1800, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 300, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1500, ICHAIN_STOP),
+};
+
+Color_RGB8 sColors[] = {
+	{0x64, 0x78, 0x8C}, {0x50, 0x8C, 0xC8}, {0x64, 0x96, 0xC8}, {0x64, 0xC8, 0xF0},
+	{0x50, 0x6E, 0x8C}, {0x46, 0xA0, 0xE1}, {0x50, 0x64, 0x82}, {0x64, 0x6E, 0xBE},
+};
+
+void ObjWarp2block_Spawn(ObjWarp2block *this, GlobalContext *globalCtx) {
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, (u16)0x8B, 
+    	this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, 
+    	0, 0, 0, 
+    	sSpawnData[(this->dyna.actor.params >> 8) & 1].params);
+
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, (u16)0x8B, 
+    	this->dyna.actor.child->world.pos.x, this->dyna.actor.child->world.pos.y, this->dyna.actor.child->world.pos.z, 
+    	0, 0, 0, 
+    	sSpawnData[(this->dyna.actor.child->params >> 8) & 1].params);
+}
+
+s32 func_80BA1ECC(ObjWarp2block *this, GlobalContext *globalCtx) {
+    f32 temp_f0;
+    Actor* temp_a3;
+    Player* player;
+    Vec3f sp20;
+    f32 temp_f2;
+
+    if (func_80043590(&this->dyna) != 0) {
+        return 0;
+    }
+
+    temp_a3 = this->dyna.actor.child;
+	player = PLAYER;
+    if ((this->dyna.actor.xzDistToPlayer <= sDistances[(((this->dyna.actor.params >> 0xB) & 7))]) || 
+    	(temp_a3->xzDistToPlayer <= sDistances[(((temp_a3->params >> 0xB) & 7))])) {
+
+        func_8002DBD0(&this->dyna.actor, &sp20, &player->actor.world.pos);
+        temp_f2 = (this->dyna.actor.scale.x * 50.0f) + 6.0f;
+
+        if (!(temp_f2 < fabsf(sp20.x)) && !(temp_f2 < fabsf(sp20.z))) {
+            return 0;
+        }
+
+        func_8002DBD0(temp_a3, &sp20, &player->actor.world.pos);
+        temp_f2 = (temp_a3->scale.x * 50.0f) + 6.0f;
+
+        if (!(temp_f2 < fabsf(sp20.x))) {
+            temp_f0 = fabsf(sp20.z);
+            if (!(temp_f2 < temp_f0)) {
+                return 0;
+            }
+        }
```

Comment:
```suggestion
        if (!(temp_f2 < fabsf(sp20.x)) && !(temp_f2 < fabsf(sp20.z))) {
            return 0;
        }
```
Does this work? Also, negated float compares here are weird, but not obvious how to fix them.

Reply:
That works but it doesn't allow me to remove temp_f0 from declarations even though it's now unused.

Yeah I tried a bunch of stuff with changing how control flowed around them in order to remove the negations, but nothing that matches.

Reply:
stack pads are pretty common. Just change it to "s32 pad"

Reply:
Sounds good! Done.